service/s3/s3manager/upload.go

```diff
@@ -508,6 +508,7 @@ func (u *multiuploader) upload(firstBuf io.ReadSeeker) (*UploadOutput, error) {
 
 	// Read and queue the rest of the parts
 	for u.geterr() == nil {
+		num++
```

Comment:
My reading of the code is that it is correct now. 

num starts on 1, because one part gets sent before the loop starts.

Say MaxUploadParts is 2.

First time around the loop num gets incremented to 2 before the check against MaxUploadParts.

The num > MaxUploadParts check succeeds and we send that part (the second).

Second time around the loop num gets incremented to 3 - the check fails and the loop is aborted before sending the 3rd part.

The test that fails without this change is `TestUploadOrderMultiBufferedReaderExceedTotalParts`.  Looking at that code I think that is correct.

It worked before because the loop did one extra iteration it shouldn't have done.

I could be wrong though so a second opinion would be good!